### PR TITLE
Pro 2418 util

### PR DESCRIPTION
**Pull request by abea**
## Summary

PRO-2418

## What are the specific steps to test this change?

Review the documentation for accuracy and clarity

## What kind of change does this PR introduce?
*(Check at least one)*

- [ ] Bug fix
- [ ] New feature
- [ ] Refactor
- [x] Documentation
- [ ] Build-related changes
- [ ] Other

## Make sure the PR fulfills these requirements:

- [x] It includes a) the existing issue ID being resolved, b) a convincing reason for adding this feature, or c) a clear description of the bug it resolves
- [x] The changelog is updated
- [x] Related documentation has been updated
- [x] Related tests have been updated

If adding a new feature without an already open issue, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

### CODE REVIEW

**Review by boutell — CHANGES_REQUESTED** [review]

File: docs/reference/modules/_template.md [hunk 1]
```diff
@@ -35,6 +35,14 @@ Curabitur blandit tempus porttitor. Nullam quis risus eget urna mollis ornare ve
 
 Etiam porta sem malesuada magna mollis euismod. Maecenas faucibus mollis interdum. Integer posuere erat a ante venenatis dapibus posuere velit aliquet.
 
+## Template helpers
+
+Template helpers are methods available for use in template files. Because this module has an alias, you can call these in templates using the alias path. For example, `apos.util.log()`.
+
+#### `slugify(string, options)`
+
+Morbi leo risus, porta ac consectetur ac, vestibulum at eros. Sed posuere consectetur est at lobortis. Lorem ipsum dolor sit amet, consectetur adipiscing elit.
```
[the remark this answers is not in the document]
Comment (abea) [reply docs/reference/modules/_template.md:44]: This is a template file providing structure for other module reference pages, hence the placeholder text.

File: docs/reference/modules/util.md [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+
+Logs an error message. The default implementation wraps `console.error` and passes on all arguments, so substitution strings may be used.
+
+### String manipulation methods
+
+#### `globalReplace(source, target, replacement)`
+
+Globally replaces a target string within a source string. This allows global string replacement without using regular expressions. Arguments are:
+
+- **`source`**: A string that will be mutated by replacing substrings.
```
[the remark this answers is not in the document]
Comment (abea) [reply docs/reference/modules/util.md:74]: This is true. I was trying to get at how this doesn't return the updated version, but my wording is bad.

File: docs/reference/modules/util.md [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+
+#### `getManagerOf(object)`
+
+Given a widget or document object, this returns the appropriate manager module.
+
+#### `get(obj, path)`
+
+Returns the value at the given `path` from the object or array `obj`.
+
+`path` supports dot notation like MongoDB. If the first dot notation segment of `path` begins with `@xyz` (or `@` followed by any alphanumeric value) the nested object within `obj` with an `_id` property equal to `xyz` is found and returned, no matter how deeply nested it is.
```
[the remark this answers is not in the document]
Comment (abea) [reply docs/reference/modules/util.md:192]: I'm trying to get at what the method returns. I actually got that phrasing from your original code comment. I don't think I understand what your replacement text is getting at.

File: docs/reference/modules/util.md [hunk 4]
```diff
@@ -0,0 +1,334 @@
...
+
+Logs the properties of an object, `obj`, in detail. Invokes the Node.js [`util.inspect()`](https://nodejs.org/docs/latest-v8.x/api/util.html#util_util_inspect_object_options) method on the object, to a depth of 10.
+
+### `reverse(arr)`
+
+Returns the array, `arr`, in reverse order.
+
+### `beginsWith(list, val)`
+
+If the `list` argument is a string, returns `true` if it starts with the value `val`. If the `list` argument is an array, returns true if *at least one* of its items begins with `val`.
```
[the remark this answers is not in the document]
Comment (abea) [reply docs/reference/modules/util.md:273]: Yeah. We don't have to document it and can mark it for deprecation.

**Review by boutell — APPROVED** [review]

**Pull request merged**